tests/unit/plugins/inventory/test_opennebula.py

```diff
@@ -266,56 +280,73 @@ def test_get_connection_info(inventory, mocker):
     assert (auth.username and auth.password)
 
 
-def test_populate_constructable_templating(inventory, mocker):
-    # bypass API fetch call
-    inventory._get_vm_pool = mocker.MagicMock(side_effect=get_vm_pool_json)
-    inventory.get_option = mocker.MagicMock(side_effect=mk_get_options(options_constructable_test))
+def test_populate_constructable_templating(mocker):
+    inventory_filename = '/fake/opennebula.yml'
+
+    mocker.patch.object(InventoryModule, '_get_vm_pool', side_effect=get_vm_pool_json)
+    mocker.patch('ansible_collections.community.general.plugins.inventory.opennebula.HAS_PYONE', True)
+    mocker.patch('ansible.inventory.manager.unfrackpath', mock_unfrackpath_noop)
+    mocker.patch('os.path.exists', exists_mock(inventory_filename))
+    mocker.patch('os.access', access_mock(inventory_filename))
 
     # the templating engine is needed for the constructable groups/vars
     # so give that some fake data and instantiate it.
-    fake_config_filepath = '/fake/opennebula.yml'
-    fake_cache = {fake_config_filepath: options_constructable_test.copy()}
-    fake_cache[fake_config_filepath]['plugin'] = 'community.general.opennebula'
-    dataloader = create_autospec(DataLoader, instance=True)
-    dataloader._FILE_CACHE = fake_cache
-    inventory.templar = Templar(loader=dataloader)
-
-    inventory._populate()
+    C.INVENTORY_ENABLED = ['community.general.opennebula']
+    inventory_file = {inventory_filename: r'''
+---
+plugin: community.general.opennebula
+api_url: https://opennebula:2633/RPC2
+api_username: username
+api_password: password
+api_authfile: '~/.one/one_auth'
+hostname: v4_first_ip
+group_by_labels: true
+compose:
+  is_linux: GUEST_OS == 'linux'
+filter_by_label: bench
+groups:
+  benchmark_clients: TGROUP.endswith('clients')
+  lin: is_linux == true
+keyed_groups:
+  - key: TGROUP
+    prefix: tgroup
+'''}
+    im = InventoryManager(loader=DictDataLoader(inventory_file), sources=inventory_filename)
```

Comment:
Generally letting ansible-core handling most of the plugin setup makes the tests more robust. I've only changed the one test that no longer works with the data tagging PR merged, so now we have at least one example in this repository of how this works.